code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+		if(mindTypeTwo == "Lower")
+			animorphsSwap()
+			return
+		if(mindTypeOne == "Higher" && theFly.len == 0)
+			higherSwap()
+			return
+	else
+		errorMessage = "Unknown interruption"
+		currentlySwapping = FALSE
+		swapProgress = 0
+		unlockPods()
+
+/obj/machinery/mind_machine/mind_machine_hub/emp_act()
+	if(currentlySwapping)
+		malfSwap = TRUE
+		return
+
+/obj/machinery/mind_machine/mind_machine_hub/emag_act()
+	if(currentlySwapping)
+		spark(src)
+		malfSwap = TRUE
+		return
+
+//Procs for the mind swap//////////
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/lowerSwap(var/mob/living/LO, var/mob/living/LT)
+	LO = occupantOne
+	LT = occupantTwo
+	if(isanimal(LO) && isanimal(LT))
+		var/mob/living/simple_animal/SO = LO
+		var/mob/living/simple_animal/ST = LT
+		var/occOneFaction = SO.faction
+		var/occTwoFaction = ST.faction
+		SO.faction = occTwoFaction //Make friendly spiders and angry corgis
+		ST.faction = occOneFaction
+		var/occOneSpeakChance = SO.speak_chance
+		var/occTwoSpeakChance = ST.speak_chance
+		SO.speak_chance = occTwoSpeakChance
+		ST.speak_chance = occOneSpeakChance
+		var/occOneName = SO.name
+		var/occTwoName = ST.name
+		SO.name = occTwoName
+		ST.name = occOneName
+		var/list/occOneSpeak = list()
+		var/list/occTwoSpeak = list()
+		for(var/V in SO.speak)
+			occOneSpeak += V
+			SO.speak -= V
+		for(var/V in ST.speak)
+			occTwoSpeak += V
+			ST.speak -= V
+		for(var/V in occTwoSpeak)
+			SO.speak += V
+		for(var/V in occOneSpeak)
+			ST.speak += V
+		var/occOnePet = ST.is_pet
+		var/occTwoPet = SO.is_pet
+		SO.is_pet = occTwoPet
+		ST.is_pet = occOnePet
+	else
+		errorMessage = "Sum intelligence too low" //There's just nothing to swap for monkeymen
+		bluespaceConduit += 1
+	currentlySwapping = FALSE
+	occupantScan = FALSE
+	connectOne.icon_state = "mind_pod_closed"
+	connectTwo.icon_state = "mind_pod_closed"
+	icon_state = "mind_hub"
+	swapProgress = 0
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/higherSwap(var/mob/HO, var/mob/HT)
+	HO = occupantOne
+	HT = occupantTwo
+	if(HO.mind.special_verbs.len)	//Swapping players, stolen from mind transfer
+		for(var/V in HO.mind.special_verbs)
+			HO.verbs -= V
+	if(HT.mind.special_verbs.len)
+		for(var/V in HT.mind.special_verbs)
+			HT.verbs -= V //Simple animals can't have special verbs so we just remove them without re-adding. Admins fear the machine.
+	var/list/HO_spells = HO.spell_list.Copy()
+	var/list/HT_spells = HT.spell_list.Copy()
+	for(var/spell/S in HO.spell_list)
+		HO.remove_spell(S)
+	for(var/spell/S in HT.spell_list)
+		HT.remove_spell(S)
+	var/mob/living/dummy = new(src.loc)
+	HO.mind.transfer_to(dummy)
+	HT.mind.transfer_to(HO)
+	dummy.mind.transfer_to(HT)
+	qdel(dummy)
+	for(var/spell/S in HO_spells)
+		HT.add_spell(S)
+	for(var/spell/S in HT_spells)
+		HO.add_spell(S)
+	HO.confused = 8
+	HO.dizziness = 8
+	HT.confused = 8
+	HT.dizziness = 8
+	currentlySwapping = FALSE
+	occupantScan = FALSE
+	connectOne.icon_state = "mind_pod_closed"
```

Comment:
You should not duplicate all these "reset state" things all over the place like you are currently.

Reply:
Do you mean the dizziness = 8 instead of += or things like occupantScan = FALSE?
The latter will be difficult to make into a re-usable proc because it's not always resetting the same things. Unlock() attempts to serve that purpose but I was only able to use it in some places.

Reply:
Stuff like the
```
currentlySwapping = FALSE
occupantScan = FALSE
connectOne.icon_state = "mind_pod_closed"
```